app/views/users/list.html.haml

```diff
@@ -1,5 +1,18 @@
 %h1 Users
 
+%h2
+  = link_to 'Add user', new_user_url
+
+%h3 
+  Filter by
+  = link_to 'All', root_url
+  |
+  = link_to 'Trim', root_url + '?action_type=trim'
```

Comment:
Please read up on http://api.rubyonrails.org/classes/ActionView/Helpers/UrlHelper.html
Adding a query string to the end is not recommended, you should instead use the :params parameter.